firrtl/src/main/scala/firrtl/ir/IR.scala

```diff
@@ -349,6 +349,13 @@ case class Conditionally(info: Info, pred: Expression, conseq: Statement, alt: S
     with HasInfo
     with UseSerializer
 
+case class DefContract(info: Info, names: Seq[String], exprs: Seq[Expression], body: Statement)
+    extends Statement
+    with HasInfo
+    with UseSerializer {
+  require(names.size == exprs.size)
```

Comment:
Please include a string providing more information about this. (It's relatively self-explanatory, admittedly...)